### PR TITLE
config-tools: modified requirements.txt and update GSG

### PR DESCRIPTION
**Pull request by zihengL1**
Moved requirements.txt form misc/config_tools/configurator
directory to misc/config_tools directory and modified
relevant python package installing steps in GSG.

Tracked-On: #7975
Reviewed-by: Junjie Mao <junjie.mao@intel.com>
Signed-off-by: Ziheng Li <ziheng.li@intel.com>

### CODE REVIEW

File: misc/config_tools/requirements.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 lxml
-elementpath
-xmlschema
+elementpath==2.5.3
+xmlschema==1.9.2
 build
 tqdm
 requests
 xmltodict
 sphinx
```
Comment (dbkinder) [inline misc/config_tools/requirements.txt:9]: The previous list (in the GSG) only had 5 python packages.  Why do we need the new ones added here: build, requests, xmltodict, and sphinx?  (If they're for the configurator, we don't have the user build the configurator any more so we don't need those in the general developer prerequisites.

We need an older version of sphinx for documentation and not just the latest available.

**Review by dbkinder — CHANGES_REQUESTED** [review]
There seem to be too many python packages as prerequisites for the GSG...

**Review by dbkinder — APPROVED** [review]
That looks better, thanks.  This looks like the right list for the general developer working on ACRN. I'm a bit concerned now that we don't have the prerequisites for building the configurator recorded any more, but that's a different problem.

@NanlinXie @zihengL1This PR removes misc/config_tools/configurator/requirements.txt
so the instructions for [building the configurator from source](https://projectacrn.github.io/latest/tutorials/acrn_configurator_tool.html#build-acrn-configurator-from-source-code) will now fail.  As for the documentation, it's OK for this special instructions to have additional dependencies, but we need to be careful not to introduce incompatible dependencies among the different tools.

**Review by NanlinXie — APPROVED** [review]

**Review by dbkinder — COMMENTED** [review]
@NanlinXie @zihengL1 We discussed not installing python packages that are not needed for developers building ACRN, yet some packages that are only needed for building the configurator got back on this list in the GSG and the PR was merged.  :(

**Comment by junjiemao1** [comment]
Mmm... Sorry for not noticing the discussion in this PR before.

Dropping dependencies of the configurator may not be a good idea if we already have one specifying its dependencies. How about going back to the original approach that maintains two requirements.txt under `misc/config_tools` and `misc/config_tools/configurator` (which was what we want to get rid of at that time, unfortunately)?

**Comment by dbkinder** [comment]
@junjiemao1 @NanlinXie Are we going to revert this change then, or leave it as merged for the 3.1 release?

**Comment by dbkinder** [comment]
@junjiemao1 @NanlinXie When doing a full GSG walkthrough, the instructions in the GSG say to checkout the v3.0 tag so when you try to use the requirements.txt file it doesn't exist (it's only on the master branch).  I submitted PR #8015 to revert the GSG to not use requirements.txt while we figure things out.  (When we make the v3.1 release, we can update to use the v3.1 tag in the GSG, but that's not ready yet.)

**Comment by junjiemao1** [comment]
> @junjiemao1 @NanlinXie Are we going to revert this change then, or leave it as merged for the 3.1 release?

I would suggest we keep it and add another PR to split it into two dependency lists under different directories if we are fine with keeping two of them.

**Pull request merged**